repos/system_upgrade/common/actors/scanthirdpartytargetpythonmodules/libraries/scanthirdpartytargetpythonmodules.py

```diff
@@ -0,0 +1,189 @@
+    return not result['exit_code']
+
+
+def identify_files_of_pypackages(syspaths):
+    ts = rpm.TransactionSet()
+    # resolving paths to absolute paths and adding trailing slash
+    roots = tuple(os.path.join(str(Path(path).resolve()), "") for path in syspaths)
+    file_to_pkg = {}
+
+    # Iterate over all installed packages
+    for header in ts.dbMatch():
+        pkg = header['name']
+        files = header['filenames']
+        for filename in files:
+            if filename and filename.endswith(PYTHON_EXTENSIONS) and filename.startswith(roots):
+                file_to_pkg[filename] = pkg
+    return file_to_pkg
+
+
+def find_python_related(root):
+    # recursively search for all files matching the given extension
+    for pattern in PYTHON_EXTENSIONS:
+        yield from Path(root).rglob("*" + pattern)
+
+
+def _should_skip_file(file):
+    # pyc files are importable, but not if they are in __pycache__
+    return file.name.endswith(".pyc") and file.parent.name == "__pycache__"
+
+
+def scan_python_files(system_paths, rpm_files):
+    """
+    Scan system paths for Python files and categorize them by ownership.
+
+    :param system_paths: List of paths to scan for Python files
+    :param rpm_files: Dictionary mapping file paths to RPM package names
+    :return: Tuple of (rpms_to_check, third_party_unowned_files) where:
+             - rpms_to_check is a dict mapping RPM names to list of their files
+             - third_party_unowned_files is a list of files not owned by any RPM
+    """
+    rpms_to_check = defaultdict(list)
+    third_party_unowned_files = []
+
+    for path in system_paths:
+        if not Path(path).is_dir():
+            continue
+        for file in find_python_related(path):
+            if _should_skip_file(file):
+                continue
+            owner = rpm_files.get(str(file))
+            if owner:
+                rpms_to_check[owner].append(file)
+            else:
+                third_party_unowned_files.append(str(file))
+
+    return rpms_to_check, third_party_unowned_files
+
+
+def identify_unsigned_rpms(rpms_to_check):
+    """
+    Identify which RPMs are third-party (not signed by the distribution).
+
+    :param rpms_to_check: Dictionary mapping RPM names to list of their files
+    :return: Tuple of (third_party_rpms, third_party_files) where:
+             - third_party_rpms is a list of third-party RPM package names
+             - third_party_files is a list of files from third-party RPMs
+    """
+    third_party_rpms = []
+    third_party_files = []
+
+    for rpm_name, files in rpms_to_check.items():
+        if not has_package(DistributionSignedRPM, rpm_name):
+            third_party_rpms.append(rpm_name)
+            api.current_logger().warning(
+                'Found Python files from non-distribution RPM package: {}'.format(rpm_name)
+            )
+            third_party_files.extend([str(f) for f in files])
```

Comment:
Files should already be a list of strings, right?
```suggestion
            third_party_files.extend(files)
```